d2/src/main/java/com/linkedin/d2/xds/XdsClientImpl.java

```diff
@@ -267,10 +267,15 @@ private void readyHandler()
         boolean cancelledTimeout = _readyTimeoutFuture.cancel(false);
         _log.info("ADS stream ready, cancelled timeout task: {}", cancelledTimeout);
         _readyTimeoutFuture = null; // set it to null to avoid repeat notifications to subscribers.
-        _xdsClientJmx.incrementReconnectionCount();
+        if (_retryRpcStreamFuture != null)
```

Comment:
Nit : Could we early returns to reduce nesting
```java
   if (!_adsStream.isReady())
    {
      return;
    }

    // if the ready timeout future is non-null, a reconnect notification hasn't been sent yet.
    if (_readyTimeoutFuture != null)
    {
      // timeout task will be cancelled only if it hasn't already executed.
      boolean cancelledTimeout = _readyTimeoutFuture.cancel(false);
      _log.info("ADS stream ready, cancelled timeout task: {}", cancelledTimeout);
      _readyTimeoutFuture = null; // set it to null to avoid repeat notifications to subscribers.
      if (_retryRpcStreamFuture != null)
      {
        _log.info("ADS stream reconnected after {} milliseconds",
                  _retryRpcStreamFuture.getDelay(TimeUnit.MILLISECONDS));
        _retryRpcStreamFuture = null;
        _xdsClientJmx.incrementReconnectionCount();
      }
      notifyStreamReconnect();
    }
```

Reply:
yeah, updated